services/api/src/api/routes/endpoint.py

```diff
@@ -28,21 +30,77 @@
     get_json_ok_response,
 )
 
+InputForProcessingStepsMapping = Mapping[str, List[ProcessingStep]]
+
+EndpointProcessingStepsMapping = Mapping[str, InputForProcessingStepsMapping]
+
 
 class EndpointsDefinition:
     """Definition of supported endpoints and its relation with processing steps."""
 
-    processing_steps_by_endpoint: Mapping[str, List[ProcessingStep]]
+    processing_steps_by_endpoint: EndpointProcessingStepsMapping
 
     def __init__(self, graph: ProcessingGraph, endpoint_config: EndpointConfig):
         self.definition = {
-            endpoint: [graph.get_step(step) for step in processing_steps]
-            for endpoint, processing_steps in endpoint_config.specification.items()
+            endpoint: {
+                input_type: [graph.get_step(step) for step in steps] for input_type, steps in input_types.items()
+            }
+            for endpoint, input_types in endpoint_config.specification.items()
         }
 
 
+@dataclass
+class InputParams:
+    input_type: str
+    dataset: str
+    config: Optional[str]
+    split: Optional[str]
+
+
+def get_params(query_params: QueryParams) -> InputParams:
+    input_type = "dataset"
+    dataset = query_params.get("dataset")
+
+    if not are_valid_parameters([dataset]) or not dataset:
+        raise MissingRequiredParameterError("Parameter 'dataset' is required")
+
+    config = None
+    config_param = query_params.get("config")
+    if are_valid_parameters([config_param]):
+        input_type = "config"
+        config = config_param
+
+    split = None
+    split_param = query_params.get("split")
+    if are_valid_parameters([split_param]):
+        input_type = "split"
+        split = split_param
+        if config is None:
+            raise MissingRequiredParameterError("Parameter 'config' is required")
+
+    return InputParams(input_type=input_type, dataset=dataset, config=config, split=split)
+
+
+def first_entry_from_steps(processing_steps: List[ProcessingStep], params: InputParams) -> Optional[CacheEntry]:
+    for processing_step in processing_steps:
+        try:
+            result = get_response(
+                kind=processing_step.cache_kind,
+                dataset=params.dataset,
+                config=params.config,
+                split=params.split,
+            )
+            return result
+        except DoesNotExist:
+            logging.warning(
+                f"processing_step={processing_step.name} dataset={params.dataset} "
+                f"config={params.config} split={params.split} no entry found"
+            )
+    return None
+
+
 def create_endpoint(
```

Comment:
(nit) would be cool to add a test for this one at one point (not particularly in this PR though)

Reply:
Agree, for now, I added test for the functions used inside this one.